lib/active_merchant/billing/gateways/pay_hub.rb

```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
+        '4508' => "INVALID_YEAR_MONTH_DAYS",
+        '4509' => "INVALID_YEAR_BILL_ON_MONTH_NO",
+        '4510' => "INVALID_MONTHLY_BILL_TYPE",
+        '4511' => "INVALID_MDAYS_POSITION",
+        '4512' => "INVALID_MDAYS",
+        '4513' => "INVALID_MDAY_EACH_POSITIONS",
+        '4514' => "INCONSISTENT_SCHEDULE_FIELDS",
+        '4515' => "ACCOUNT_IS_NOT_ADMINISTRATOR",
+        '4516' => "INCONSISTENT_CARD_DATA_FIELDS",
+        '4517' => "WRONG_OAUTH_TOKEN_FOR_MERCHANT",
+        '4518' => "NO_MERCHANT_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4519' => "NO_SETUP_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4520' => "NO_VT_CLIENT_ACCOUNT_FOR_OUTH_TOKEN_FOR_MERCHANT",
+        '4521' => "INVALID_CURRENCY_CODE",
+        '4522' => "INVALID_RECURRING_BILL_STATUS_CHANGE_CODE",
+        '4523' => "INCONSISTENT_CURRENCIES",
+        '4524' => "MISSING_PROPERTY_CODE",
+        '4525' => "INCONSISTENT_CUSTOMER_DATA_CODE",
+        '4526' => "DUPLICATE_DEVELOPER_CODE",
+        '9989' => "METHOD_REQUEST_BODY_VALIDATION_ERROR_CODE",
+        '9990' => "HTTP_NOT_READABLE_EXCEPTION_CODE",
+        '9991' => "JSONPARSE_EXCEPTION_CODE",
+        '9992' => "MAPPING_EXCEPTION_CODE",
+        '9993' => "MEDIA_TYPE_NOT_SUPPORTED_CODE",
+        '9994' => "METHOD_NOT_SUPPORTED_CODE",
+        '9995' => "PAGE_NOT_FOUND_ERROR_CODE",
+        '9996' => "ACCESS_DENIED_ERROR_CODE",
+        '9997' => "JSON_SYNTAX_ERROR_CODE",
+        '9998' => "NOT_FOUND_ERROR_CODE",
+        '9999' => "INTERNAL_SERVER_ERROR_CODE"
       }
 
       def initialize(options={})
-        requires!(options, :orgid, :username, :password, :tid)
-
+        requires!(options, :organization_id, :oauth_token, :terminal_id, :version)
         super
       end
-
-      def authorize(amount, creditcard, options = {})
-        post = setup_post('auth')
-        add_creditcard(post, creditcard)
-        add_amount(post, amount)
-        add_address(post, (options[:address] || options[:billing_address]))
+      
+      def purchase(amount, creditcard, options={})
+        post = setup_post(options)
+        add_bill(post, amount, options)
+        add_creditcard(post, creditcard, options)
         add_customer_data(post, options)
-
-        commit(post)
+        
+        commit(:post, "sale", post, options)
       end
 
-      def purchase(amount, creditcard, options={})
-        post = setup_post('sale')
-        add_creditcard(post, creditcard)
-        add_amount(post, amount)
-        add_address(post, (options[:address] || options[:billing_address]))
+      def recurring(amount, creditcard, options={})
+        post = setup_post(options)
+        add_bill(post, amount, options)
+        add_creditcard(post, creditcard, options)
         add_customer_data(post, options)
-
-        commit(post)
+        add_schedule(post, options)
+        
+        commit(:post, "recurring-bill", post, options)
       end
 
-      def refund(amount, trans_id, options={})
```

Comment:
@j-mutter 

Actually we are no longer using `amount` in refund api, but we can add for maintaining signature, Please guide me.